django/db/models/sql/query.py

```diff
@@ -331,10 +342,8 @@ def clone(self):
         obj.used_aliases = self.used_aliases.copy()
         obj._filtered_relations = self._filtered_relations.copy()
         # Clear the cached_property
-        try:
-            del obj.base_table
-        except AttributeError:
-            pass
+        if 'base_table' in obj.__dict__:
+            obj.__dict__.pop('base_table')
```

Comment:
Could we simply

```suggestion
        obj.__dict__.pop('base_table', None)
```

To avoid the double lookup?

Reply:
Ah, a good point, thanks. I think the `None` part was me starting to go down that road, only to (I think) get distracted by the need to show the line-profile timings and be thorough.

I'll be sure to change it, if there's a general consensus of opinion that this overall demonstrates some merit.